.github/workflows/ci.yml

```diff
@@ -489,11 +571,18 @@ jobs:
           testing/bin/ptl suite run \
             --suite ${{ matrix.suite }} --config config-${{ matrix.config }} --bind=off --logs-dir logs/ --timeout 2h \
             --trino-jdk-version zulu_${{ matrix.jdk }}
-      - name: Upload test results
+      - name: Upload test logs and results
         uses: actions/upload-artifact@v2
         if: failure()
         with:
-          name: result pt (${{ matrix.config }}, ${{ matrix.suite }})
+          name: result pt (${{ matrix.config }}, ${{ matrix.suite }}, ${{ matrix.jdk }})
           path: |
             presto-product-tests/target/*
             logs/*
+      - name: Upload test report
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: test report pt (${{ matrix.config }}, ${{ matrix.suite }}, ${{ matrix.jdk }})
+          path: presto-product-tests/target/reports/testng-results.xml
```

Comment:
seems redundant with respect to `presto-product-tests/target/*` above.

also `presto-product-tests` -> `testing/trino-product-tests` ??

Reply:
The difference between those two steps is that the other one grabs all test reports so the size of the artifact is greater, and it only runs on failure. I added this as a separate job so it can run always and produce much smaller artifact (with shorter retention).

I'll rename the test results path in a separate commit.

Reply:
Done, added comments about separate test uploads to every job.